From ccd76805007be4f91f9802cdb889df1be9f57afa Mon Sep 17 00:00:00 2001
From: Aymeric PLOTON <aymeric.ploton@provenrun.com>
Date: Thu, 13 Jul 2023 18:35:55 +0200
Subject: [PATCH 2/5] Add support for ProvenCore

ProvenCore requires secure SGIs to be handled at S-EL1. This patch
overrides the default ZynqMP configuration to handle them at EL3 in case
ProvenCore SPD is enabled.

%% original patch: 0003-feat-zynqmp-add-support-for-ProvenCore.patch
From: Florian LUGOU <florian.lugou@provenrun.com>
Date: 2021-09-07 11:49:58

From: Florian LUGOU <florian.lugou@provenrun.com>
Date:
---
 plat/xilinx/zynqmp/platform.mk | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/plat/xilinx/zynqmp/platform.mk b/plat/xilinx/zynqmp/platform.mk
index 99a4beb32..7919e2379 100644
--- a/plat/xilinx/zynqmp/platform.mk
+++ b/plat/xilinx/zynqmp/platform.mk
@@ -1,5 +1,6 @@
 #
 # Copyright (c) 2013-2021, ARM Limited and Contributors. All rights reserved.
+# Portions copyright (c) 2021-2022, ProvenRun S.A.S. All rights reserved.
 #
 # SPDX-License-Identifier: BSD-3-Clause
 
@@ -11,9 +12,19 @@ SEPARATE_CODE_AND_RODATA := 1
 ZYNQMP_WDT_RESTART := 0
 IPI_CRC_CHECK := 0
 override RESET_TO_BL31 := 1
-override GICV2_G0_FOR_EL3 := 1
 override WARMBOOT_ENABLE_DCACHE_EARLY := 1
 
+# pncd SPD requires secure SGI to be handled at EL1
+ifeq (${SPD},pncd)
+	ifeq (${ZYNQMP_WDT_RESTART},1)
+		$(error "Error: ZYNQMP_WDT_RESTART and SPD=pncd are incompatible")
+	endif
+	override GICV2_G0_FOR_EL3 := 0
+else
+	override GICV2_G0_FOR_EL3 := 1
+endif
+
+
 # Do not enable SVE
 ENABLE_SVE_FOR_NS	:= 0
 
-- 
2.34.1

